File: python/matrix/matrix.py
# coding=utf-8
from __future__ import unicode_literals

__author__ = 'Cedric Zhuang'


class Matrix(object):
    def __init__(self, raw):
        self.rows = [list(map(int, row.split()))
                     for row in raw.split('\n')]
        self.columns = list(map(list, zip(*self.rows)))

File: scala/matrix/src/main/scala/Matrix.scala
/**
  *
  * @author Cedric Zhuang
  *
  */
case class Matrix(repr: String) {
  lazy val rows: Vector[Vector[Int]] =
    repr.split("\n").map(_.split(" ").map(_.toInt).toVector).toVector

  lazy val cols: Vector[Vector[Int]] = rows.transpose.map(_.toVector)
}
